azurecompute/src/main/java/org/jclouds/azurecompute/compute/functions/DeploymentToNodeMetadata.java

```diff
@@ -16,16 +16,99 @@
+	private final Map<String, Credentials> credentialStore;
+
+	public static final Map<Deployment.Status, NodeMetadata.Status> serverStateToNodeStatus = ImmutableMap
+			  .<Deployment.Status, NodeMetadata.Status> builder()
+			  .put(Deployment.Status.DELETING, NodeMetadata.Status.PENDING)
+			  .put(Deployment.Status.SUSPENDED_TRANSITIONING, NodeMetadata.Status.PENDING)
+			  .put(Deployment.Status.RUNNING_TRANSITIONING, NodeMetadata.Status.PENDING)
+			  .put(Deployment.Status.DEPLOYING, NodeMetadata.Status.PENDING)
+			  .put(Deployment.Status.STARTING, NodeMetadata.Status.PENDING)
+			  .put(Deployment.Status.SUSPENDED, NodeMetadata.Status.SUSPENDED)
+			  .put(Deployment.Status.RUNNING, NodeMetadata.Status.RUNNING)
+			  .put(Deployment.Status.UNRECOGNIZED, NodeMetadata.Status.UNRECOGNIZED).build();
+
+	@Inject
+	DeploymentToNodeMetadata(@Memoized Supplier<Set<? extends Location>> locations,
+										GroupNamingConvention.Factory namingConvention, OSImageToImage osImageToImage,
+										RoleSizeToHardware roleSizeToHardware, Map<String, Credentials> credentialStore) {
+		this.nodeNamingConvention = checkNotNull(namingConvention, "namingConvention").createWithoutPrefix();
+		this.locations = checkNotNull(locations, "locations");
+		this.osImageToImage = checkNotNull(osImageToImage, "osImageToImage");
+		this.roleSizeToHardware = checkNotNull(roleSizeToHardware, "roleSizeToHardware");
+		this.credentialStore = checkNotNull(credentialStore, "credentialStore cannot be null");
+	}
+
 	@Override
-	public NodeMetadata apply(Deployment input) {
-		return null;
+	public NodeMetadata apply(Deployment from) {
+		NodeMetadataBuilder builder = new NodeMetadataBuilder();
+		builder.ids(from.name());
+		builder.name(from.name());
+		builder.hostname(getHostname(from));
+		/* TODO
+		if (from.getDatacenter() != null) {
+			builder.location(from(locations.get()).firstMatch(
+					  LocationPredicates.idEquals(from.getDatacenter().getId() + "")).orNull());
+		}
+		builder.group(nodeNamingConvention.groupInUniqueNameOrNull(from.getHostname()));
+		builder.hardware(roleSizeToHardware.apply(from.instanceSize()));
+		Image image = osImageToImage.apply(from);
+		if (image != null) {
+			builder.imageId(image.getId());
+			builder.operatingSystem(image.getOperatingSystem());
+		}
+		*/
+		if (from.status() != null) {
+			builder.status(serverStateToNodeStatus.get(from.status()));
+		}
+		Set<String> publicIpAddresses = Sets.newLinkedHashSet();
+		if (from.virtualIPs() != null) {
+			for (Deployment.VirtualIP virtualIP : from.virtualIPs()) {
+				publicIpAddresses.add(virtualIP.address());
+			}
+			builder.publicAddresses(publicIpAddresses);
+		}
+		Set<String> privateIpAddresses = Sets.newLinkedHashSet();
+		if (from.roleInstanceList() != null) {
+			for (Deployment.RoleInstance roleInstance : from.roleInstanceList()) {
+				privateIpAddresses.add(roleInstance.ipAddress());
+			}
+			builder.privateAddresses(privateIpAddresses);
+		}
```

Comment:
Does the API return the credentials? If not, consider setting there from the credential store as a best-effort to get them. Take [DigitalOcean](https://github.com/jclouds/jclouds-labs/blob/master/digitalocean/src/main/java/org/jclouds/digitalocean/compute/functions/DropletToNodeMetadata.java#L115-L120) as an example.